pkg/scheduler/queue/user_queues.go

```diff
@@ -237,6 +245,9 @@ func (q *queues) getNextQueueForQuerier(lastUserIndex int, querierID string) (us
 			continue
 		}
 
+		q.userQueuesMx.RLock()
```

Comment:
this line is not covered by the unit test

Reply:
modified the test to cover this line